ApiDemos/java/app/build.gradle

```diff
@@ -76,11 +80,12 @@ task generateV3(type: Copy) {
 }
 
 task generateV3Layout(type: Copy) {
+    group "V3 Beta"
+    description "Copies layout files from GMS to V3 BETA."
+
     from 'src/gms/res/layout'
     into 'src/v3/res/layout'
```

Comment:
If we follow a naming convention for all V3 specific examples, or put them in their own package, you could still run this on every build and avoid overwriting v3 specific examples by adding an exclusion line above the `filter` like:

```groovy
exclude { details -> details.file.name.equals('*v3.*') }
```

I've done this in another project, although with a single file, to keep an implementation-specific class that differs for each product flavor, while the rest of the classes are copied and the import package changed.

It would be nice to have the source code changes for both sets in every PR - if the task is only run occasionally, the changesets will likely be out of sync, and CI won't be able to catch potential issues when the code is changed.

Reply:
I like that rule—perhaps v3 specific files can be in its own package. There could also be changes though that differ on the same named file in which case this task will overwrite the change in V3 when it shouldn't. A more sophisticated task is certainly needed here but I'd prefer to address that separately.